WordPress/Classes/ViewRelated/Reader/Select Interests/Mock Data/InterestsDataSource.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+class ReaderInterest {
+    var isSelected: Bool = false
+
+    let title: String
+    let slug: String
+
+    init(title: String, slug: String) {
+        self.title = title
+        self.slug = slug
+    }
+}
+
+class InterestsDataSource {
+    private(set) var count: Int = 0
+    private(set) var interests: [ReaderInterest] = []
+
+    init(fileName: String) {
+        parseJSON(fileName: fileName)
+    }
+
+    private func parseJSON(fileName: String) {
+        guard
```

Comment:
Since this is a brand new code I highly suggest using [Decodable](https://developer.apple.com/documentation/swift/decodable) to parse JSON, we'll be able to remove a loot of lines here.

Reply:
This is going to be removed in the future.